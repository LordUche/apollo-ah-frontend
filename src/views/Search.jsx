import React from 'react';
import { PropTypes } from 'prop-types';
import ArticleBox from './ArticleBox';
import CommentLoader from './CommentLoader';


export default function Search({
  result, handleSearch, q, categoryId, handleChange
}) {
  const { articles, page, loading } = result;

  let resultPanel = <CommentLoader />;
  if (!loading) {
    resultPanel = !articles || articles.length === 0
      ? (
        <header className="page-header page-header-sm text-center">
          <h4> No articles match your search result...</h4>
        </header>
      )
      : (
        <>
          <header className="page-header page-header-sm">
            <h1>
              {' '}
              Showing
              {' '}
              {page.description}
              {' '}
            </h1>
          </header>
          <div className="row">
            {articles.map(article => <ArticleBox article={article} key={article.id} />)}
          </div>
        </>
      );
  }


  return (
    <main className="main-body">
      <section className="search-cover">
        <div className="container">
          <form className="search" onSubmit={handleSearch} noValidate>
            <div className="row">
              <div className="col-sm-6 col-8">
                <i className="fas fa-search search-icon" />
                <input className="search-input no-focus" name="q" value={q} onChange={handleChange} />
              </div>
              <div className="col-sm-3 d-none d-sm-inline-block">
                <select className="custom-select search-select no-focus" name="categoryId" value={categoryId} onChange={handleChange}>
                  <option value={0}>All categories</option>
                  <option value={3}>Business</option>
                  <option value={7}>Entertainment</option>
                  <option value={4}>Fashion</option>
                  <option value={6}>Food</option>
                  <option value={5}>Health</option>
                  <option value={8}>History</option>
                  <option value={1}>Technology</option>
                  <option value={2}>Sports</option>
                  <option value={9}>Others</option>
                </select>
              </div>
              <div className="text-right col-sm-3 col-4">
                <button type="submit" className="btn btn-brand search-button shadow-none">Search</button>
              </div>
            </div>
          </form>
        </div>
      </section>
      <section className="wrapper min-vh-100 ">
        <div className="container">
          {resultPanel}
        </div>
      </section>

    </main>
  );
}

Search.propTypes = {
  result: PropTypes.object.isRequired,
  handleSearch: PropTypes.func.isRequired,
  q: PropTypes.string,
  categoryId: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
};
Search.defaultProps = {
  q: '',
  categoryId: '0',
};
