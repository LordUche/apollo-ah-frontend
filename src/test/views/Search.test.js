import React from 'react';
import { shallow } from 'enzyme';
import { createMockStore } from '../setup';
import Search from '../../views/Search';

createMockStore();

describe('<Search>', () => {
  it('should render', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toMatchSnapshot();
  });
});
