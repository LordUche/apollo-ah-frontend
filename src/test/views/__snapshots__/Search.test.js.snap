// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search> should render 1`] = `
<main
  className="main-body"
  data-pg-collapsed={true}
>
  <section
    className="search-cover"
    data-pg-collapsed={true}
  >
    <div
      className="container"
    >
      <div
        className="search"
      >
        <div
          className="row"
          data-pg-collapsed={true}
        >
          <div
            className="col-sm-6 col-8"
            data-pg-collapsed={true}
          >
            <i
              className="fas fa-search search-icon"
            />
            <input
              className="search-input no-focus"
            />
          </div>
          <div
            className="col-sm-3 d-none d-sm-inline-block"
            data-pg-collapsed={true}
          >
            <select
              className="custom-select search-select no-focus"
              data-pg-collapsed={true}
              id="category"
              name="category"
            >
              <option
                selected={true}
                value={0}
              >
                All categories
              </option>
              <option
                value={3}
              >
                Business
              </option>
              <option
                value={7}
              >
                Entertainment
              </option>
              <option
                value={4}
              >
                Fashion
              </option>
              <option
                value={6}
              >
                Food
              </option>
              <option
                value={5}
              >
                Health
              </option>
              <option
                value={8}
              >
                History
              </option>
              <option
                value={1}
              >
                Technology
              </option>
              <option
                value={2}
              >
                Sports
              </option>
              <option
                value={9}
              >
                Others
              </option>
            </select>
          </div>
          <div
            className="text-right col-sm-3 col-4"
          >
            <button
              className="btn btn-brand search-button shadow-none"
              type="button"
            >
              Search
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    className="wrapper min-vh-100 "
    data-pg-collapsed={true}
  >
    <div
      className="container"
    >
      <header
        className="page-header page-header-sm"
      >
        <h1>
          Showing 1-7 of 45
           
        </h1>
        <div
          className="btn-group btn-group-toggle float-right view-changer"
          data-toggle="buttons"
        >
          <label
            className="btn active"
          >
            <input
              autoComplete="off"
              defaultChecked={true}
              name="options"
              type="radio"
            />
            <i
              className="fas fa-th"
            />
          </label>
          <label
            className="btn"
          >
            <input
              autoComplete="off"
              name="options"
              type="radio"
            />
            <i
              className="fas fa-list"
            />
          </label>
        </div>
      </header>
      <div
        className="row view-row"
      >
        <div
          className="col-sm-6 col-lg-4"
          data-pg-collapsed={true}
        >
          <div
            className="article"
          >
            <a
              className="link-surface"
              href="single-article.html"
            />
            <div
              className="article-cover"
            >
              <img
                className="article-cover-img"
                src="assets/images/articles/thought-catalog-470985-unsplash.jpg"
              />
            </div>
            <div
              className="article-item"
            >
              <span
                className="article-category"
              >
                Health
              </span>
              <h4>
                5 ways to manage your time effectively.
              </h4>
              <p>
                Distinctively coordinate pandemic technologies rather than market-driven meta-services. Distinctively reconceptualize high standards in...
              </p>
              <div
                className="article-author"
              >
                <img
                  src="assets/images/authors/photo-1524154217857-45f012d0f167.jpg"
                />
                <div>
                  <h6>
                    Alexandra
                  </h6>
                  <span>
                    22 hrs ago
                  </span>
                  <span
                    className="article-author-dot"
                  />
                  <span>
                    4 min read
                  </span>
                  <i
                    className="fas fa-bookmark transition"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-sm-6 col-lg-4"
          data-pg-collapsed={true}
        >
          <div
            className="article transition"
          >
            <a
              className="link-surface"
              href="single-article.html"
            />
            <div
              className="article-cover"
            >
              <img
                className="article-cover-img transition"
                src="assets/images/articles/brooke-lark-158017-unsplash.jpg"
              />
            </div>
            <div
              className="article-item"
            >
              <span
                className="article-category"
              >
                Food
              </span>
              <h4>
                5 ways to manage your time effectively.
              </h4>
              <p>
                Distinctively coordinate pandemic technologies rather than market-driven meta-services. Distinctively reconceptualize high standards in...
              </p>
              <div
                className="article-author"
              >
                <img
                  src="assets/images/authors/photo-1524154217857-45f012d0f167.jpg"
                />
                <div>
                  <h6>
                    Alexandra
                  </h6>
                  <span>
                    22 hrs ago
                  </span>
                  <span
                    className="article-author-dot"
                  />
                  <span>
                    4 min read
                  </span>
                  <i
                    className="fas fa-bookmark transition"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-sm-6 col-lg-4"
          data-pg-collapsed={true}
        >
          <div
            className="article transition"
          >
            <a
              className="link-surface"
              href="single-article.html"
            />
            <div
              className="article-cover"
            >
              <img
                className="article-cover-img transition"
                src="assets/images/articles/avel-chuklanov-509630-unsplash.jpg"
              />
            </div>
            <div
              className="article-item"
            >
              <span
                className="article-category"
              >
                Entertainment
              </span>
              <h4>
                5 ways to manage your time effectively.
              </h4>
              <p>
                Distinctively coordinate pandemic technologies rather than market-driven meta-services. Distinctively reconceptualize high standards in...
              </p>
              <div
                className="article-author"
              >
                <img
                  src="assets/images/authors/photo-1524154217857-45f012d0f167.jpg"
                />
                <div>
                  <h6>
                    Alexandra
                  </h6>
                  <span>
                    22 hrs ago
                  </span>
                  <span
                    className="article-author-dot"
                  />
                  <span>
                    4 min read
                  </span>
                  <i
                    className="fas fa-bookmark transition bookmarked"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-sm-6 col-lg-4"
          data-pg-collapsed={true}
        >
          <div
            className="article transition"
          >
            <a
              className="link-surface"
              href="single-article.html"
            />
            <div
              className="article-cover"
            >
              <img
                className="article-cover-img transition"
                src="assets/images/articles/hammer-tusk-79752-unsplash.jpg"
              />
            </div>
            <div
              className="article-item"
            >
              <span
                className="article-category"
              >
                Technology
              </span>
              <h4>
                5 ways to manage your time effectively.
              </h4>
              <p>
                Distinctively coordinate pandemic technologies rather than market-driven meta-services. Distinctively reconceptualize high standards in...
              </p>
              <div
                className="article-author"
              >
                <img
                  src="assets/images/authors/photo-1524154217857-45f012d0f167.jpg"
                />
                <div>
                  <h6>
                    Alexandra
                  </h6>
                  <span>
                    22 hrs ago
                  </span>
                  <span
                    className="article-author-dot"
                  />
                  <span>
                    4 min read
                  </span>
                  <i
                    className="fas fa-bookmark transition bookmarked"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-sm-6 col-lg-4"
          data-pg-collapsed={true}
        >
          <div
            className="article transition"
          >
            <a
              className="link-surface"
              href="single-article.html"
            />
            <div
              className="article-cover"
            >
              <img
                className="article-cover-img transition"
                src="assets/images/articles/rawpixel-558596-unsplash.jpg"
              />
            </div>
            <div
              className="article-item"
            >
              <span
                className="article-category"
              >
                Business
              </span>
              <h4>
                5 ways to manage your time effectively.
              </h4>
              <p>
                Distinctively coordinate pandemic technologies rather than market-driven meta-services. Distinctively reconceptualize high standards in...
              </p>
              <div
                className="article-author"
              >
                <img
                  src="assets/images/authors/photo-1524154217857-45f012d0f167.jpg"
                />
                <div>
                  <h6>
                    Alexandra
                  </h6>
                  <span>
                    22 hrs ago
                  </span>
                  <span
                    className="article-author-dot"
                  />
                  <span>
                    4 min read
                  </span>
                  <i
                    className="fas fa-bookmark transition"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-sm-6 col-lg-4"
          data-pg-collapsed={true}
        >
          <div
            className="article transition"
          >
            <a
              className="link-surface"
              href="single-article.html"
            />
            <div
              className="article-cover"
            >
              <img
                className="article-cover-img transition"
                src="assets/images/articles/tim-gouw-68319-unsplash.jpg"
              />
            </div>
            <div
              className="article-item"
            >
              <span
                className="article-category"
              >
                Technology
              </span>
              <h4>
                5 ways to manage your time effectively.
              </h4>
              <p>
                Distinctively coordinate pandemic technologies rather than market-driven meta-services. Distinctively reconceptualize high standards in...
              </p>
              <div
                className="article-author"
              >
                <img
                  src="assets/images/authors/photo-1524154217857-45f012d0f167.jpg"
                />
                <div>
                  <h6>
                    Alexandra
                  </h6>
                  <span>
                    22 hrs ago
                  </span>
                  <span
                    className="article-author-dot"
                  />
                  <span>
                    4 min read
                  </span>
                  <i
                    className="fas fa-bookmark transition"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <ul
        className="pagination justify-content-center mt-5"
      >
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
          >
            <i
              className="fas fa-chevron-left"
            />
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
          >
            1
          </a>
        </li>
        <li
          className="page-item active"
        >
          <a
            className="page-link active"
            href="#"
          >
            2
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
          >
            3
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
          >
            <i
              className="fas fa-chevron-right"
            />
          </a>
        </li>
      </ul>
    </div>
  </section>
</main>
`;
